### DISCUSSION TITLE
Get Number of Breakpoints?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [mfussenegger/nvim-dap#810] [Q&A] [answered]
Is it possible to get the number of set breakpoints, e.g. for the statusline or for other plugins to know? In the docs, I only found `list_breakpoints`, which adds them to the quickfix list, instead of giving an output.

**Comment by mfussenegger** [comment, marked as answer]
There is currently no documented way, but you could use `require('dap.breakpoints').get()` to get all breakpoints grouped by buffer and then count them.

Just keep in mind that the API is considered internal and may change without going through deprecation. 
There won't be a public breakpoint API until there's support for all breakpoint features (data breakpoints and function breakpoints are still missing)

**Reply by chrisgrieser** [reply]
great, thanks, I'll keep that in mind!

**Reply by chrisgrieser** [reply]
just for the context: I am working [on a plugin that adds breakpoints to macros](https://github.com/chrisgrieser/nvim-recorder). The idea I had was to use shared keymaps for DAP and the macro-plugin: 
- `addBreakPoint` will map to `dap.toggle_breakpoint()` outside a recording. During a recording, it will add a macro breakpoint instead.
- `playMacro` will map to `dap.continue()` if there is at least one dap-breakpoint. If there is no dap breakpoint, will play the macro instead

The idea is to save people the mental overhead of learning too many new keymaps. will flag that feature as experimental until the DAP API for this is stable / non-internal.
